pkg/jar/parse.go

```diff
@@ -0,0 +1,468 @@
+package jar
+
+import (
+	"archive/zip"
+	"bufio"
+	"bytes"
+	"crypto/sha1"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+
+	"go.uber.org/zap"
+	"golang.org/x/xerrors"
+
+	"github.com/aquasecurity/go-dep-parser/pkg/log"
+	"github.com/aquasecurity/go-dep-parser/pkg/types"
+)
+
+const (
+	baseURL         = "http://search.maven.org/solrsearch/select"
+	idQuery         = `g:"%s"+AND+a:"%s"`
+	artifactIdQuery = `a:"%s"+AND+p:"jar"`
+	sha1Query       = `1:"%s"`
+)
+
+var (
+	jarFileRegEx = regexp.MustCompile(`^([a-zA-Z0-9\._-]*[^-*])-(\d\S*(?:-SNAPSHOT)?).jar$`)
+)
+
+type conf struct {
+	baseURL      string
+	rootFilePath string
+}
+
+type Option func(*conf)
+
+func WithURL(url string) Option {
+	return func(c *conf) {
+		c.baseURL = url
+	}
+}
+
+func WithFilePath(filePath string) Option {
+	return func(c *conf) {
+		c.rootFilePath = filePath
+	}
+}
+
+func Parse(r io.Reader, opts ...Option) ([]types.Library, error) {
+	c := conf{baseURL: baseURL}
+	for _, opt := range opts {
+		opt(&c)
+	}
+
+	return parseArtifact(c, c.rootFilePath, ioutil.NopCloser(r))
```

Comment:
If input jar file for Parse method,  fileName get "." string.

https://github.com/aquasecurity/go-dep-parser/pull/16/files#diff-97e351ec9e6a7af74afd305e76098e1d4a25f64297ea478ab8e150d48617ab60R79